core/src/main/scala/org/apache/spark/ui/storage/StorageTab.scala

```diff
@@ -59,7 +59,15 @@ class StorageListener(storageStatusListener: StorageStatusListener) extends Bloc
 
   override def onStageSubmitted(stageSubmitted: SparkListenerStageSubmitted): Unit = synchronized {
     val rddInfos = stageSubmitted.stageInfo.rddInfos
-    rddInfos.foreach { info => _rddInfoMap.getOrElseUpdate(info.id, info) }
+    rddInfos.foreach { info =>
```

Comment:
I see. Actually, you can just use one line code  `rddInfos.foreach { info => _rddInfoMap.getOrElseUpdate(info.id, info).name = info.name }`